packages/venia-concept/src/actions/app.js

```diff
@@ -1,15 +1,30 @@
-import timeout from 'src/util/timeout';
-import {
-    drawerClose as closeMs,
-    drawerOpen as openMs
-} from 'src/shared/durations';
-
-export const toggleDrawer = drawerName => dispatch => {
-    dispatch({
-        type: 'TOGGLE_DRAWER',
-        payload: drawerName
-    });
-    return timeout(drawerName ? openMs : closeMs);
-};
-
-export const closeDrawer = () => toggleDrawer(null);
+import { createActions } from 'redux-actions';
+
+import { store } from 'src';
+
+const prefix = 'APP';
+const actionTypes = ['TOGGLE_DRAWER'];
+
+const actions = createActions(...actionTypes, { prefix });
+export default actions;
+
+/* async action creators */
+
+export const loadReducers = payload =>
+    async function thunk() {
+        try {
+            const reducers = await Promise.all(payload);
+
+            reducers.forEach(({ default: reducer, name }) => {
+                store.addReducer(name, reducer);
+            });
+        } catch (error) {
+            console.log(error);
+        }
+    };
```

Comment:
Async action creators aren't actually action creators at all. They don't return an action object with a `type`, and they aren't handled by reducers. They're just higher-order functions useful for orchestrating dispatches.

Here, `loadReducers` doesn't even dispatch any actions. Should we still call it an async action creator? or a helper? Is there a difference, and does it matter?

Reply:
When a reducer is added, the store actually replaces its root reducer, which instructs connected components to re-render (like an action does). And there's nothing that a reducer should do in response to `REDUCERS_ADDED`; even updating the list of current slices is pointless, since `getState()` would be the source of truth for that info.

I think we can improve the reducer-adding workflow—turn it into a pipeline, maybe—as we open up to extensibility. But for now I think this is actually fine.